src/Plugin/CustomerController/AjaxLogin.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare(strict_types=1);
+
+namespace Hyva\Checkout\Plugin\CustomerController;
+
+use Hyva\Checkout\Config\Checkout as HyvaCheckoutConfig;
+use Hyva\Checkout\Model\SectionData\LoginSectionDataServiceFactory;
+use Magento\Customer\Controller\Ajax\Login as CustomerAjaxLoginController;
+use Hyva\Checkout\Result\Json as JsonResult;
+use Magento\Framework\Controller\ResultInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Ajax login will authenticate user, merge carts and correctly set session.
+ * Doing the above using graphql is a pain actually where we need to authenticate
+ * user, merge carts and then finally fetch cart info. Still, will face session
+ * issues. So rely on ajax login is better here.
+ */
+class AjaxLogin
+{
+    /**
+     * @var \Hyva\Checkout\Config\Checkout
+     */
+    private $hyvaCheckoutConfig;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \Hyva\Checkout\Model\SectionData\LoginSectionDataServiceFactory
+     */
+    private LoginSectionDataServiceFactory $loginSectionDataServiceFactory;
+
+    /**
+     * AjaxLogin constructor.
+     *
+     * @param  \Psr\Log\LoggerInterface  $logger
+     * @param  \Hyva\Checkout\Config\Checkout  $hyvaCheckoutConfig
+     * @param  LoginSectionDataServiceFactory  $loginSectionDataServiceFactory
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        HyvaCheckoutConfig $hyvaCheckoutConfig,
+        LoginSectionDataServiceFactory $loginSectionDataServiceFactory
```

Comment:
Yes, I used the factory to avoid the dependencies in the service class loading unnecessarily. we need that to happen only when we have a valid json response from the controller